packages/react/src/components/Toggletip/__tests__/Toggletip-test.js

```diff
@@ -4,23 +4,183 @@
  * This source code is licensed under the Apache-2.0 license found in the
  * LICENSE file in the root directory of this source tree.
  */
+import React, { forwardRef } from 'react';
+import { fireEvent, render, screen } from '@testing-library/react';
+import { Toggletip, ToggletipButton } from '..';
+import userEvent from '@testing-library/user-event';
 
 describe('Toggletip', () => {
   describe('accessibility', () => {
     test.todo('accessibility-checker');
-    test.todo('axe');
+    it('should have no Axe violations', async () => {
+      const { container } = render(
+        <Toggletip data-testid="toggletip">test</Toggletip>
+      );
+
+      await expect(container).toHaveNoAxeViolations();
+    });
   });
 
   // Usage
-  it.todo('should toggle visibility on click');
-  it.todo('should toggle visibility on Enter and Space');
-  it.todo('should close on Escape');
-  it.todo('should close if an element outside of the toggletip is clicked');
+  it('should toggle visibility on click', () => {
+    const { container } = render(
+      <Toggletip data-testid="toggletip">
+        <ToggletipButton label="Show information">test</ToggletipButton>
+      </Toggletip>
+    );
+    userEvent.click(screen.getByText('test'));
+
+    expect(screen.getByRole('button')).toHaveAttribute('aria-expanded', 'true');
+    expect(container.firstChild).toHaveClass('cds--toggletip--open');
+    expect(container.firstChild).toHaveClass('cds--popover--open');
+  });
```

Comment:
you are going to have to make all functions that use `userEvent` async and then await the action
```suggestion
  it('should toggle visibility on click', async () => {
    const { container } = render(
      <Toggletip data-testid="toggletip">
        <ToggletipButton label="Show information">test</ToggletipButton>
      </Toggletip>
    );
    await userEvent.click(screen.getByText('test'));

    expect(screen.getByRole('button')).toHaveAttribute('aria-expanded', 'true');
    expect(container.firstChild).toHaveClass('cds--toggletip--open');
    expect(container.firstChild).toHaveClass('cds--popover--open');
  });
```